### PR TITLE
Prepare repair for task manager integration

### PR DESCRIPTION
**Pull request by Deexie**
The PR prepares repair for task manager integration:
- Creates repair_module
- Keeps repair_module in repair_service
- Moves tracker methods to repair_module
- Changes UUID to task_id in repair module

### CODE REVIEW

File: api/api-doc/task_manager.json [hunk 1]
```diff
@@ -49,6 +49,14 @@
                         "type":"string",
                         "paramType":"path"
                     },
+                    {
```
Comment (Deexie) [inline api/api-doc/task_manager.json:52]: #11821

File: tasks/task_manager.hh [hunk 2]
```diff
@@ -14,13 +14,13 @@
 #include <seastar/coroutine/parallel_for_each.hh>
```
Comment (Deexie) [inline tasks/task_manager.hh:14]: #11821

File: api/task_manager_test.cc [hunk 3]
```diff
@@ -52,7 +52,7 @@ void set_task_manager_test(http_context& ctx, routes& r, db::config& cfg) {
         it = req->query_parameters.find("entity");
```
Comment (Deexie) [inline api/task_manager_test.cc:52]: #11821

File: tasks/task_manager.hh [hunk 4]
```diff
@@ -318,37 +318,37 @@ public:
             co_await _gate.close();
```
Comment (Deexie) [inline tasks/task_manager.hh:319]: #11821

**Review by bhalevy — COMMENTED** [review]
LGTM overall.
Some mnor comments and room for further cleanups as follow up when moving more functionality to the generic task manager classes.

File: repair/repair_task.hh [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+ */
+
+#pragma once
+
+#include "repair/repair.hh"
+#include "tasks/task_manager.hh"
+
+class repair_task_impl : public tasks::task_manager::task::impl {
+public:
+    repair_task_impl(tasks::task_manager::module_ptr module, tasks::task_id id, unsigned sequence_number = 0, std::string keyspace = "", std::string table = "", std::string type = "", std::string entity = "", tasks::task_id parent_id = tasks::task_id::create_null_id())
```
Comment (denesb) [inline repair/repair_task.hh:16]: Why so many default parameters. I noticed this being a repeating pattern in task related code. Is there some user of this API which is significantly helped by all this?

File: repair/repair.hh [hunk 6]
```diff
@@ -64,7 +64,15 @@ struct repair_uniq_id {
     // The integer ID used to identify a repair job. It is currently used by nodetool and http API.
     int id;
     // A UUID to identifiy a repair job. We will transit to use UUID over the integer ID.
-    tasks::task_id uuid;
+    tasks::task_info task_data;
```
Comment (denesb) [inline repair/repair.hh:71]: The comment above needs an update.

File: repair/repair.cc [hunk 7]
```diff
@@ -1002,7 +1001,7 @@ int repair_service::do_repair_start(sstring keyspace, std::unordered_map<sstring
     // nothing to repair, and return 0. "nodetool repair" prints in this case
     // that "Nothing to repair for keyspace '...'". We don't have such a case
     // yet. Real ids returned by next_repair_command() will be >= 1.
-    auto id = repair_tracker().next_repair_command();
```
Comment (denesb) [inline repair/repair.cc:1006]: Comment above needs update.

**Review by denesb — COMMENTED** [review]
LGTM, some small comments.

**Comment by Deexie** [comment]
- rebased on next
- deleted default parameters from `repair_task_impl`
- updated comments concerning `repair_uniq_id`
- `task_data` -> `task_info` in `repair_uniq_id`

**Comment by denesb** [comment]
@Deexie int he future, please don't set the target branch to `next`, leave it as `master`.

**Pull request merged**